src/etc/inc/filter.inc

```diff
@@ -1169,6 +1169,7 @@ function filter_generate_optcfg_array() {
 						$oic['vips6'][$vipidx]['sn'] = $vip['subnet_bits'];
 					}
 				}
+                if (isset($vip['mode'])) $oic['vips'][$vipidx]['mode']=$vip['mode'];
```

Comment:
Please use [K&R, BSD KNF variant style](https://doc.pfsense.org/index.php/Developer_Style_Guide#Indent_style), even for simple statements, to keep them consistent and easy to read, and be careful to use matching and proper indentation.